### ISSUE TITLE
Error Out of Bounds

### ISSUE THREAD

**Issue by vpatton** [gemini-testing/gemini#394]
I get this from time to time running my tests on saucelabs, mostly IE  9 - Edge

Output of `gemini --version`:
2.1.1
...

Contents of `.gemini.yml` file:

``` yaml
rootUrl: http://beerfridayautotest.cms.dealer.com
tolerance: 3
screenshotsDir: './screens'
windowSize: 1600x1080
sessionsPerBrowser: 5
suitesPerSession: 10
calibrate: true

browsers:
  IE-10:
    screenshotsDir: './screens/ie10-screens'
    desiredCapabilities:
      version: "10.0"
      browserName: 'internet explorer'
      platform: "Windows 8"
      screenResolution: "1280x1024"
      name: "IE10 BeerFriday"

  IE-9:
    screenshotsDir: './screens/ie-screens'
    desiredCapabilities:
      version: "9.0"
      browserName: 'internet explorer'
      platform: "Windows 7"
      screenResolution: "1280x1024"
      name: "IE9 BeerFriday"

  EdgeWin10:
    screenshotsDir: './screens/edge'
    desiredCapabilities:
      version: "20.10240"
      browserName: 'MicrosoftEdge'
      platform: "Windows 10"
      screenResolution: "1280x1024"
      name: "EdgeWin10 BeerFriday"

system:
  plugins:
    sauce:
      username:
      accessKey:
```

Test source code:

``` js
    gemini.suite('beerfridayautotest Inventory Search', function (el) {
        el.setCaptureElements('div.ddc-container.tabbed-widgets-group.tabbed-widgets-group-1.tabs-slide-direction-left')
        .capture('Inventory Search', function (actions, el){
            actions.waitForElementToShow('div.ddc-container.tabbed-widgets-group.tabbed-widgets-group-1.tabs-slide-direction-left', 5000)
        });
    });
```

Command used to run the test:

```
gemini test ./test/ -c sauce.yml --reporter html
```

Publically accessible URL of the page you are testing against:

https://beerfridayautotest.cms.dealer.com/

Result:
Error: Out of the bounds
    at Error (native)
    at module.exports.inherit.fill (/Users/van.patton/test-framework-gemini/node_modules/png-img/index.js:66:19)
    at module.exports.inherit.fill (/Users/van.patton/test-framework-gemini/node_modules/png-img/index.js:56:25)
    at clear (/Users/van.patton/test-framework-gemini/node_modules/gemini/lib/image.js:31:19)
    at /Users/van.patton/test-framework-gemini/node_modules/gemini/lib/capture-session.js:74:25
    at Array.forEach (native)
    at module.exports.inherit._cropImage (/Users/van.patton/test-framework-gemini/node_modules/gemini/lib/capture-session.js:73:33)
    at wrapper (/Users/van.patton/test-framework-gemini/node_modules/gemini/node_modules/lodash/index.js:3533:19)
    at _fulfilled (/Users/van.patton/test-framework-gemini/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/van.patton/test-framework-gemini/node_modules/q/q.js:863:30)
...

Expected behaviour:
Hopefully a screenshot
...

**Comment by SwinX** [comment]
@vpatton hi, sorry for delayed response.

From log you provided it seems that you set some ignore selectors in your tests. Are them really set and if so what are they?

**Comment by vpatton** [comment]
`gemini.suite('autotest About page', function (suite) {
    suite.setUrl('/dealership/about.htm?_disablePixall=true')
        .before(function (actions, find) {
            actions.wait(3000)
        })
        .setCaptureElements('div.ddc-content.header-value-statement')
        .ignoreElements('.cycle.ddc-list-unstyled')
        .capture('header');
    gemini.suite('autotest Navigation Menu', function (el) {
        el.setCaptureElements('div[data-widget-id="template-navigation1"]')
        .capture('Navigation Menu', function (actions, el){
            actions.waitForElementToShow('div[data-widget-name="navigation-default"]', 5000)
        });
    });
});`

**Comment by hukka** [comment]
I've seen the same issue happen, when the ignored element is not (fully) contained within the bounds of the captured element.

**Comment by sipayRT** [comment]
so you want to ignore element, but it positioned outside of the page?

**Comment by hukka** [comment]
Not outside the page, but outside the captured element. Partly outside is enough too.

**Comment by hannesj** [comment]
These three images explain our problem. The black box hides a span containing some text and an animated svg that sticks a couple of pixels above the selection. We would like to hide just the animated svg, but can't due to this issue. Now it also makes the tests to fail sometimes, as the animation is slightly different above the captured area.

![chrome50 current](https://cloud.githubusercontent.com/assets/698100/17210681/ee848792-54cc-11e6-9b10-44c684e929c2.png)
![chrome50 ref](https://cloud.githubusercontent.com/assets/698100/17210682/f082a4de-54cc-11e6-8ac5-62b0df06aa2b.png)
![chrome50 diff](https://cloud.githubusercontent.com/assets/698100/17210686/f2e521c0-54cc-11e6-822d-ec6d299e758a.png)

**Comment by sipayRT** [comment]
@hannesj you can add wrap for your selected block that will be with pudding. So your svg never will be outside of the box

**Comment by varya** [comment]
I have the same problem. I ignore elements which are not seen at some dimensions, and so get an error. Why should the test fall in this case? The element I would like to ignore is not seen, so it should be OK.

**Comment by sipayRT** [comment]
@varya https://github.com/gemini-testing/gemini/issues/543

**Comment by sipayRT** [comment]
> The element I would like to ignore is not seen, so it should be OK.

why should you ignore element if it doesn't seen on the page?

**Comment by andreyb06** [comment]
I am having the same issue trying to ignore a dynamic element that changes with every page load. The css element is not actually present in the UI but loads outside and I'm getting the Out of the bounds errors trying to ignore that element. Any way around it?

**Comment by zapur1** [comment]
I'm also seeing this, were the ignored element is on the page but not part of the first screenshot of the page

**Comment by leevh** [comment]
Also seeing same issue as previous two posts.  Any workaround on later versions?

**Comment by Dmitriy-kiselyov** [comment]
Fixed in 5.4.1
[Proof](https://github.com/gemini-testing/gemini/blob/master/CHANGELOG.md#541-2018-02-12)
